python_modules/dagster/dagster/_cli/utils.py

```diff
@@ -1,35 +1,68 @@
+import logging
 import os
 import tempfile
 from contextlib import contextmanager
-from typing import Callable, Iterator, Optional
-
-import click
+from typing import Iterator, Optional
 
 from dagster._core.instance import DagsterInstance, InstanceRef
 from dagster._core.instance.config import is_dagster_home_set
+from dagster._core.secrets.env_file import get_env_var_dict
 
 
 @contextmanager
-def get_instance_for_service(
-    service_name: str,
+def get_instance_for_cli(
+    cli_command: Optional[str] = "this command",
     instance_ref: Optional[InstanceRef] = None,
-    logger_fn: Callable[[str], None] = click.echo,
+    fallback_to_tempdir: bool = False,
+    logger: logging.Logger = logging.getLogger("dagster"),
 ) -> Iterator[DagsterInstance]:
+    """Create an instance at the entrypoint for a dagster CLI command. Handles loading
+    any environment variables from a local .env file and finding the right path to load
+    the instance from.
+
+    Parameters:
+    cli_command: The name of the command - optional, used for logging messages.
+    instance_ref: For commands that delegate to other commands, they can pass in
+        a serialized instance_ref object rather than loading it from a dagster.yaml file.
+    fallback_to_tempdir: What the command should do if DAGSTER_HOME is not set and no InstanceRef
+        is passed in. If true, will create a temporary instance using the local filesystem that is
+        removed when the process ends.
+    logger: Customize any log messages emitted while creating the instance.
+    """
+    dotenv_file_path = os.getcwd()
+    env_var_dict = get_env_var_dict(dotenv_file_path)
+
+    try:
+        env_var_dict = get_env_var_dict(dotenv_file_path)
+    except:
+        logger.exception("Failure loading environment variables from .env file")
+        env_var_dict = {}
```

Comment:
not a big fan of eating all exceptions here. At a minimum would should log the inner exception. I'd argue that we should just hard fail if there is a syntax error that makes the file unparseable, to avoid confusion.

Reply:
Actually i think if that was an issue it would be an issue before this PR too, so it should be fine to not try/catch here